### PR TITLE
Implementations template

### PR DESCRIPTION
**Pull request by dlacoste-esrf**
Add a template to help develop new backends.

### CODE REVIEW

File: example/libhdbpp-example/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Changed
+
+- Removed public headers as the only needed one is AbstractDB from libhdbpp.
```
Comment (gscalam) [inline example/libhdbpp-example/CHANGELOG.md]: CHANGELOG.md looks copy-pasted from libhdbpp-timescale, it should be cleaned

File: example/libhdbpp-example/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+find_package(libhdbpp)
+
+# If not installed then get them
+if(NOT libhdbpp_FOUND)
+    include(FetchContent)
+
+    FetchContent_Declare(
+      libhdbpp
+      GIT_REPOSITORY https://github.com/tango-controls-hdbpp/libhdbpp.git
+      GIT_TAG        project-build
```
Comment (gscalam) [inline example/libhdbpp-example/CMakeLists.txt]: GIT_TAG should be master since we're merging

File: example/libhdbpp-example/CHANGELOG.md (hunk 1 above)
Comment (dlacoste-esrf) [reply example/libhdbpp-example/CHANGELOG.md]: oupsie… You caught me, thanks !

File: example/libhdbpp-example/CMakeLists.txt (hunk 2 above)
Comment (dlacoste-esrf) [reply example/libhdbpp-example/CMakeLists.txt]: Wasn't sure about this one indeed, as for the current template to work we need project-build. But let's go for master directly !

Comment (gscalam) [reply example/libhdbpp-example/CMakeLists.txt]: It's a committment to merge soon all pending PR.

File: example/libhdbpp-example/src/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.11)
+
+# source files
+set(LOCAL_SRC_FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/HdbppExampleDbApi.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/HdbppExampleDb.cpp
```
Comment (gscalam) [inline example/libhdbpp-example/src/CMakeLists.txt:6]: missing ')'

File: example/libhdbpp-example/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+This is an empty implementation of libhdbpp to be used as a template when developping a new backend.
+
+## Bug Reports + Feature Requests
+
+Please file the bug reports and feature requests in the issue tracker
+
+## Documentation
+
+* See the [doc](doc/README.md) directory for documentation on various topics, including building and installing.
```
Comment (gscalam) [inline example/libhdbpp-example/README.md:21]: Broken doc link, remove link or create empty doc/README.md ?

Comment (dlacoste-esrf) [reply example/libhdbpp-example/README.md:21]: Added an almost empty doc folder to fix the link. I'd rather encourage people (me first) to write documentation !

**Pull request merged**